app/models/user.rb

```diff
@@ -2,8 +2,11 @@
 
 class User < ApplicationRecord
   has_many :machines, dependent: :destroy
-  has_many :subscriptions, dependent: :destroy
   has_many :free_accesses, dependent: :destroy
+  has_many :sales_as_client, class_name: 'Sale', foreign_key: 'client_id', dependent: :destroy, inverse_of: :client
+  has_many :sales_as_seller, class_name: 'Sale', foreign_key: 'seller_id', dependent: :nullify, inverse_of: :seller
+  has_many :refunds, foreign_key: 'refunder_id', dependent: :destroy, inverse_of: :refunder
+  has_many :subscriptions, through: :sales_as_client, dependent: :destroy
```

Comment:
Do we need to keep anything regarding legal/financial when a user is deleted?

Reply:
We only need to keep the invoices for ten years which are not deleted when destroying a user.